src/redux/rockets/rocketSlice.js

```diff
@@ -1,18 +1,50 @@
-import { createSlice } from '@reduxjs/toolkit'
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const apiURL = "https://api.spacexdata.com/v3/rockets";
+const FETCH_ROCKETS = "FETCH_ROCKETS";
+
+const initialState = {
+  rockets: [],
+  isLoading: true,
+};
+
+export const fetchRockets = createAsyncThunk(
+  FETCH_ROCKETS,
+  async (thunkAPI) => {
+    try {
+      const rockets = await axios.get(apiURL);
+      return rockets.data;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(error.response.data);
+    }
+  }
+);
 
 // Slice
 
 const slice = createSlice({
-    name: 'rockets',
-    initialState: {
-        rockets: []
+  name: "rockets",
+  initialState,
+  reducers: {
+    [fetchRockets.pending]: (state) => {
+      state.isLoading = true;
     },
-    reducers: {
-        getRockets: (state, action) => {
-            state.rockets = action.payload;
-        },
+    [fetchRockets.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      action.payload.map((rocket) => {
+        return (state.rockets = {
+          id: rocket.id,
+          rocket_name: rocket.rocket_name,
+          description: rocket.description,
+          images: rocket.flickr_images,
+        });
```

Comment:
Good implementation here but only the last rocket is being updated in the state.
Please update it to store all the fetched rockets into the redux store.

Reply:
Thank you for the review. i will work on it